File: cast/src/main/java/com/cinelist/ms/cast/config/FeignClientConfiguration.java
package com.cinelist.ms.cast.config;

import com.cinelist.ms.cast.handlers.RetrieveMessageErrorDecoder;
import feign.codec.ErrorDecoder;
import org.springframework.context.annotation.Bean;

public class FeignClientConfiguration {

    @Bean
    public ErrorDecoder errorDecoder() {
        return new RetrieveMessageErrorDecoder();
    }
}

File: accountskt/src/main/kotlin/com/cinelits/ms/accountskt/config/FeignClientConfiguration.kt
package com.cinelits.ms.accountskt.config

import com.cinelits.ms.accountskt.handlers.RetrieveMessageErrorDecoder
import org.springframework.context.annotation.Bean

class FeignClientConfiguration {

    @Bean
    fun errorDecoder(): RetrieveMessageErrorDecoder {
        return RetrieveMessageErrorDecoder()
    }
}